### PR TITLE
SWATCH-3018 Create a swatch-unleash-importer clowdapp

### PR DESCRIPTION
**Pull request by mstead**
The swatch-unleash-importer clowdapp defines a job that will load an unleash instance with the defined feature flags used by the subscription watch components.

# NOTES
* This work had been previously done by awood, however, this patch makes some tweaks to allow the image to be built in the standard quay repo and gets it deployed automatically into the ephemeral environment.
* Required because there is currently no way to do this directly via app-interface for the ephemeral environment.

# Testing

## Bonfire Locally
1. Build the images
```
./bin/build-images.sh -p swatch-unleash-import
```

2. Deploy via bonfire (REPLACE **$YOUR_IMAGE_TAG** WITH THE ONE FROM YOUR BUILD)
```
bonfire deploy rhsm  --timeout=1800  --optional-deps-method none  \
--frontends false  --no-remove-resources app:rhsm \
-C rhsm -p rhsm/SWATCH_UNLEASH_IMPORT_IMAGE=quay.io/mstead/swatch-unleash-import \
-p rhsm/SWATCH_UNLEASH_IMPORT_IMAGE_TAG=$YOUR_IMAGE_TAG
```

3. Use port forwarding to get access to the Unleash console.
```
oc port-forward pods/$(oc get pods | cut -d' ' -f1 | grep 'env.*featureflags') 4243:4242
```

4. Access unleash UI to ensure that the **swatch.swatch-metrics-hbi.emit-events** flag was created.
```
# user/pass: admin / unleash4all
http://localhost:4243
```

## PR Check Test
1. While the PR checks are running, look at the logs 'PR build' task to find the EE that it is running in. Search for the bonfire namespace.
2. Log into EE and use the project pertaining to your namespace.
```
oc project YOUR_EE_NAMESPACE
```
3. Use port forwarding to gain access to the unleash UI using the same credentials above.
4. Verify the feature flag is in place.

### CODE REVIEW

File: deploy/rhsm-clowdapp.yaml [hunk 1]
```diff
@@ -12,6 +12,25 @@ parameters:
...
+  - name: SWATCH_UNLEASH_IMPORT_IMAGE_TAG
+    value: latest
+  - name: SWATCH_UNLEASH_IMPORT_MEMORY_REQUEST
+    value: 500Mi
+  - name: SWATCH_UNLEASH_IMPORT_MEMORY_LIMIT
+    value: 800Mi
+  - name: SWATCH_UNLEASH_IMPORT_CPU_REQUEST
+    value: 350m
+  - name: SWATCH_UNLEASH_IMPORT_CPU_LIMIT
+    value: 500m
```
Comment (awood) [inline deploy/rhsm-clowdapp.yaml:26]: You can drop these quite a bit I think.  Check out the changes to the curl containers I made in #3740

File: deploy/rhsm-clowdapp.yaml [hunk 2]
```diff
@@ -12,6 +12,25 @@ parameters:
     value: quay.io/cloudservices/rhsm-subscriptions
   - name: IMAGE_TAG
     value: latest
+  - name: SWATCH_UNLEASH_IMPORT_IMAGE
+    value: quay.io/cloudservices/swatch-unleash-import
+  - name: SWATCH_UNLEASH_IMPORT_IMAGE_TAG
+    value: latest
+  - name: SWATCH_UNLEASH_IMPORT_MEMORY_REQUEST
+    value: 10Mi
```
Comment (lindseyburnett) [inline deploy/rhsm-clowdapp.yaml:20]:
```suggestion
    value: 25Mi
```

I was getting `forbidden: memory max limit to request ratio per Pod is 2, but provided ratio is 5.000000` and had to drop the `--no-remove-resources app:rhsm` from the bonfire command to get it to work.  Probably just need to

**Review by lindseyburnett — CHANGES_REQUESTED** [review]

File: deploy/rhsm-clowdapp.yaml (hunk 2 above)
Comment (mstead) [reply deploy/rhsm-clowdapp.yaml]: DONE.

I bumped it and had no issues. Please confirm.

**Review by lindseyburnett — APPROVED** [review]

**Pull request merged**